ocaml/xapi/xapi_database_backup.ml

```diff
@@ -0,0 +1,184 @@
+module D=Debug.Make(struct let name="xapi_database_backup" end)
+open D
+
+open Http
+open Xmlrpc_client
+open Db_cache_types
+open Stdext
+open Threadext
+
+type stat  = {created : int64; modified : int64; deleted : int64;    } [@@deriving rpc]
+type field = {fldname : string; stat : stat; value : Schema.Value.t; } [@@deriving rpc]
+type row   = {objref  : string; stat : stat; value : field list;     } [@@deriving rpc]
+type table = {tblname : string; stat : stat; value : row list;       } [@@deriving rpc]
```

Comment:
minor stylistic: I don't think we usually align `}` like this, if you add a new field you'll have to realign everything again.
We usually put each field on its own line instead.

Reply:
The style below leaves room to grow and for comments. Keeping at semicolon at the front is a Haskell-ism that is spreading in the OCaml community.
```
type field = 
  { fldname :   string    (** room for comment *)
  ; stat :      stat
  ; value :     Schema.Value.t
  } [@@deriving rpc]
```